app/scripts/ui.js

```diff
@@ -325,14 +338,21 @@ function initializeUi(activeTab, connectionStream, cb) {
   });
 }
 
-async function displayCriticalError(errorKey, err, metamaskState) {
-  const html = await getErrorHtml(errorKey, SUPPORT_LINK, metamaskState);
+async function displayCriticalError(
+  errorKey,
+  err,
+  metamaskState,
+  backgroundConnection,
+) {
+  const html = await getErrorHtml(errorKey, SUPPORT_LINK, metamaskState, err);
 
   container.innerHTML = html;
 
+  registerDesktopErrorActions(backgroundConnection, browser);
```

Comment:
Should we be code fencing the desktop specific UI code also? So functions like this and also moving the desktop components into a `desktop` folder so they'll be excluded from the other build types?

Reply:
Yes I'll add the desktop code fencing everywhere... but we should not be moving the components to a specific desktop dir, because that would cause issues when we release under flask (as anything within `**/desktop/**` will only be available on desktop build type)